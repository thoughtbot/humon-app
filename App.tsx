import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import MapView from 'react-native-maps';
import {SafeAreaView, StyleSheet} from 'react-native';

const App = () => {
  return (
    <NavigationContainer>
      <SafeAreaView style={styles.container}>
        <MapView testID="mapView" style={styles.mapContainer} />
      </SafeAreaView>
    </NavigationContainer>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  mapContainer: {
    width: '100%',
    height: '100%',
  },
});

export default App;
