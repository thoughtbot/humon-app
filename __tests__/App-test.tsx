/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const sleep = (ms: number) =>
  new Promise(resolve => setTimeout(() => resolve(true), ms));

it('renders correctly', async () => {
  const component = renderer.create(<App />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  await sleep(10);
});
