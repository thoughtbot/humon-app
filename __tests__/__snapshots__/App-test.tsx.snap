// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    testID="mapView"
  />
</RCTSafeAreaView>
`;
